### PR TITLE
Add recorder-napari recipe

### PR DESCRIPTION
**Pull request by goanpeca**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- ~[ ] If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Depends on:
- https://github.com/conda-forge/staged-recipes/pull/19224
- https://github.com/conda-forge/staged-recipes/pull/19222

### CODE REVIEW

**Comment by goanpeca** [comment]
We are in the process of creating conda packages for [napari plugins](https://www.napari-hub.org/) and related dependencies. Once this recipe passes all the checks, every time a new release goes into PyPI, a PR will be created automatically, with usually very minor need for changes so the process of maintaining this moving forward is small. Also the @conda-forge/napari team will be available to help as needed.

You can read a bit more about conda-forge at https://conda-forge.org/#contribute

--- 

@mattersoflight, do you agree with being added as a maintainer?

**Comment by mattersoflight** [comment]
It is great to hear that napari and napari plugins will be distributed via conda. Please add  @talonchandler and myself as maintainers.

**Comment by goanpeca** [comment]
@mattersoflight it seems one of the commands being tested is currently failing with:

```bash
recOrder.reconstruct --help
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/recorder-napari_1657145343057/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/recOrder.reconstruct", line 10, in <module>
Usage: recOrder.reconstruct [OPTIONS]

  parse command line arguments and return class with the arguments
    sys.exit(main())
  File "/home/conda/staged-recipes/build_artifacts/recorder-napari_1657145343057/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/lib/python3.10/site-packages/recOrder/scripts/run_pipeline.py", line 36, in main
    if Args.config:
AttributeError: 'int' object has no attribute 'config'
```

**Comment by talonchandler** [comment]
Thanks for doing this @goanpeca!

Our current help page is accessible via a non-standard command: `recOrder.help`

I am planning to change the "help" location to the more standard `recOrder --help` and `recOrder.reconstruct --help` for the next release (among many other changes). 

If this (or other) issues exclude `recOrder` from `conda-forge` that's fine for now. We're fine with inclusion on subsequent releases. 

Thanks!

**Comment by goanpeca** [comment]
@talonchandler, no need to block it, we can comment that test for now :)

**Comment by goanpeca** [comment]
On another note, I think that currently the package is adding `docs` and `examples` to this package as modules, and that is not standard (and probably not a good idea), being able to do `import docs` or `import examples` @talonchandler.

I will add a patch to avoid that in the meantime.

File: recipes/recorder-napari/modules.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
++++ b/setup.cfg
+@@ -56,6 +56,12 @@ install_requires =
+ 	napari-ome-zarr>=0.3.2
+ 	qtpy
+ 
++[options.packages.find]
++exclude =
++    examples*
++    docs*
++    recOrder.tests*
```
Comment (goanpeca) [inline recipes/recorder-napari/modules.patch:13]: I suggest adding the following upstream :) @talonchandler

**Comment by goanpeca** [comment]
This one is ready for review @conda-forge/staged-recipes :)

File: recipes/recorder-napari/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    - qtpy
+    - scipy >=1.3.0
+    - superqt >=0.2.4
+    - tqdm >=4.61.1
+    - waveorder >=1.0.0b0
+    - zarr >=2.6.1
+
+test:
+  imports:
+    - recOrder
```
Comment (jaimergp) [inline recipes/recorder-napari/meta.yaml:50]:
```suggestion
    - recOrder
    - recOrder.io
    - recOrder.compute
    - recOrder.pipelines
    - recOrder.postproc
    - recOrder.preproc
    - recOrder.scripts.run_pipeline
    - recOrder.scripts.convert_tiff_to_zarr
    - recOrder.scripts.md5_check_sum
```

**Pull request merged**